pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -102,6 +104,16 @@ type Reconciler struct {
 	configStore       configStore
 	timeoutHandler    *reconciler.TimeoutSet
 	metrics           *Recorder
+
+	// The clock for tracking time
+	clock clock.Clock
+
+	// PipelineRuns that the controller will check its TTL and attempt to delete when the TTL expires.
+	queue workqueue.RateLimitingInterface
+
+	// ListerSynced returns true if the TaskRun store has been synced at least once.
+	// Added as a member to the struct to allow injection for testing.
+	ListerSynced cache.InformerSynced
```

Comment:
im wondering if there is some way we can avoid adding more attributes and functionality to the reconciler object itself - do you think it might be possible to create a separate object for managing the timeouts, which we could use and test in isolation?
if so then maybe one way to break up #1429 (which might be what you're already going for!) would be to have this PR add that object, complete with unit test coverage, and describe how it is intended to be hooked up (I think this is pretty close to what you're doing already, it's just a bit hard right now to see the overall structure)

Reply:
Yes, in [#1429](https://github.com/tektoncd/pipeline/pull/1429#issue-328355087), I just do this in a new file[ ttl_controller.go](https://github.com/tektoncd/pipeline/blob/6256db57a92afef7f49715f5753accbc7cbfe3d4/pkg/apis/pipeline/v1alpha1/ttl_controller.go), but for convenience, I add these new attibutes in Reconciler object. I'll consider it again.